modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_render_rule_condition.js

```diff
@@ -388,7 +388,26 @@ AUI.add(
 
 					if (fieldName.match('-condition-first-operand')) {
 						var operatorSelected = instance._getOperator(index);
-						var type = instance._getDataType(field.getValue(), options);
+
+						var type;
+
+						if (instance._getType(field.getValue(), options) === 'date') {
+							type = 'date';
+							instance._conditions[index + '-condition-second-operand-input-text'].set('value', '');
+							instance._conditions[index + '-condition-second-operand-numeric'].set('value', '');
+						}
+						else {
+							type = instance._getDataType(field.getValue(), options);
+
+							if (instance._getType(field.getValue(), options) === 'numeric') {
```

Comment:
Here too,If you don't want to compare the type of the data, you just need to use two equals (==), because is more performative. But let me know your thinking.